tally_ho/apps/tally/views/super_admin.py

```diff
@@ -157,6 +158,38 @@ def get_results_duplicates(tally_id):
     return result_forms_founds
 
 
+def get_result_form_with_duplicate_results(ballot=None, tally_id=None):
+    """Build a list of result forms sorted by ballot of results forms for
+    which there are more than 1 result form in the same ballot with the same
+    number of votes per candidate, and the `duplicate_reviewed`
+    column is false.
+
+    :returns A list of result forms in the system with duplicate results.
+    """
+    duplicate_results = ResultForm.objects.exclude(results=None).values(
+        'ballot',
+        'results__votes',
+        'results__candidate').annotate(
+        duplicate_count=Count('id')).filter(
+        duplicate_count__gt=1,
+        tally_id=tally_id,
+        duplicate_reviewed=False)
+
+    if ballot:
+        duplicate_results = duplicate_results.filter(ballot=ballot)
+
+    results_form_duplicates = []
+
+    if duplicate_results:
+        for result in duplicate_results:
```

Comment:
instead of a loop do this with a single fetch of all the result forms filtered on the list of IDs